nevergrad/functions/unit_commitment/core.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import nevergrad as ng
+from ..base import ExperimentFunction
+
+class UnitCommitmentProblem1(ExperimentFunction):
+    """Model that uses conventional implementation for emi-continuous variables
+    The model is adopted from Pyomo model 1: Conventional implementation for emi-continuous variables
+    (https://jckantor.github.io/ND-Pyomo-Cookbook/04.06-Unit-Commitment.html)
+    The constraints are added to the objective with a heavy penalty for violation.
+
+    Parameters
+    ----------
+    T_points: int
+        number of time points
+    N_generators: int
+        number of generators
+    penalty_weight: float
+        weight to penalize for violation of constraints
+    """
+    def __init__(self, T_points: int = 13,
+                 N_generators: int = 3,
+                 penalty_weight : float = 10000
+                 ) -> None:
+        params = {x: y for x, y in locals().items() if x not in ["self", "__class__"]}  # for copying
+        # Demand for certain time period
+        self.T_points = T_points
+        self.demands = np.array([100 + 100*np.random.uniform() for t in range(self.T_points)])
+
+        # Generating units
+        self.N_generators = N_generators
+        self.p_max = 2*max(self.demands) / self.N_generators
+        self.p_min = 0.6*self.p_max
+
+        # Cost
+        self.cost_a = np.array([0.5 + 0.2*np.random.randn() for n in range(self.N_generators)])
+        self.cost_b = np.array([10*np.random.uniform() for n in range(self.N_generators)])
+
+        self.penalty_weight = penalty_weight
+
+        param_x = ng.p.Array(shape=(self.N_generators, self.T_points)).set_bounds(0, self.p_max)
+        param_u = ng.p.Array(shape=(self.N_generators, self.T_points)).set_bounds(0, 1).set_integer_casting()
```

Comment:
is that either 1 or 0? I feel this will probably go very wrong with this parametrization because that does not leave a lot of options.
I would consider `Choice([0, 1], repetitions=timepoints * generators)` instead but that would require reshaping later on.
@teytaud how would you deal with this?

Reply:
Thanks for the suggestion. I have made the changes accordingly. But are these statement handled differently in the backend?-

Reply:
@pacowong  Yes they are handled differently in the backend.